lib/SILOptimizer/Transforms/SILLowerAggregateInstrs.cpp

```diff
@@ -144,14 +153,19 @@ static bool expandDestroyAddr(DestroyAddrInst *DA) {
   if (Type.isAddressOnly(Module))
     return false;
 
+  bool expand = shouldExpand(Module, Type.getObjectType());
+
   // If we have a non-trivial type...
   if (!Type.isTrivial(Module)) {
     // If we have a type with reference semantics, emit a load/strong release.
     LoadInst *LI = Builder.createLoad(DA->getLoc(), Addr,
                                       LoadOwnershipQualifier::Unqualified);
     auto &TL = Module.getTypeLowering(Type);
-    TL.emitLoweredDestroyValue(Builder, DA->getLoc(), LI,
-                               TypeLowering::LoweringStyle::Deep);
+    if (expand)
+      TL.emitLoweredDestroyValue(Builder, DA->getLoc(), LI,
```

Comment:
This you could do a similar thing using perhaps a refactored helper method on LoweringStyle or TypeLowering. In this case, since emitLoweredDestroyValue is already correct, you would just get:

```
TL.emitLoweredDestroyValue(Builder, DA->GetLoc(), LI, getLoweringStyle(expand));
```